src/fe-common/irc/fe-modes.c

```diff
@@ -147,10 +147,11 @@ static void msg_multi_mode(IRC_CHANNEL_REC *channel, const char *sender,
 		rec = g_new0(MODE_REC, 1);
 		modes = g_slist_append(modes, rec);
 
+		rec->level = level;
 		rec->channel = channel;
 		rec->mode = g_strdup(mode);
 	}
-
+	rec->level = ((rec->level | level) & MSGLEVEL_ALL) | (rec->level & level);
```

Comment:
What does this line do?

Reply:
it is supposed to combine the levels in the right way: the levels (everything below MSGLEVEL_ALL) are combined whereas the flags (anything above) must all match